### PR TITLE
Map set terrain ts fix

### PR DESCRIPTION
**Pull request by CraigglesO**
fixes https://github.com/maplibre/maplibre-gl-js/issues/2992

## Launch Checklist

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Link to related issues.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [x] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

File: src/ui/map.ts [hunk 1]
```diff
@@ -1963,13 +1963,13 @@ export class Map extends Camera {
...
      */
-    setTerrain(options: TerrainSpecification): this {
+    setTerrain(options?: TerrainSpecification): this {
         this.style._checkLoaded();
 
         // clear event handlers
         if (this._terrainDataCallback) this.style.off('data', this._terrainDataCallback);
 
-        if (!options) {
+        if (options === undefined) {
```
Comment (HarelM) [inline src/ui/map.ts:1972]: This can be set to null as well.

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@
 - Correct declared return type of `Map.getLayer()` and `Style.getLayer()` to be `StyleLayer | undefined` to match the documentation ([#2969](https://github.com/maplibre/maplibre-gl-js/pull/2969))
 - Correct type of `Map.addLayer()` and `Style.addLayer()` to allow adding a layer with an embedded source, matching the documentation ([#2966](https://github.com/maplibre/maplibre-gl-js/pull/2966))
 - Throttle map resizes from ResizeObserver to reduce flicker ([#2986](https://github.com/maplibre/maplibre-gl-js/pull/2986))
+- Correct function `Map.setTerrain(options: TerrainSpecification): Map` to be `Map.setTerrain(options?: TerrainSpecification): Map` ([#2993](https://github.com/maplibre/maplibre-gl-js/pull/2993))
```
Comment (HarelM) [inline CHANGELOG.md:12]: This needs an update, or removed completely.

**Comment by CraigglesO** [comment]
interestingly the test data uses undefined. I'll add both real quick, but let me know if you'd prefer the test data to be more precise to null.

File: CHANGELOG.md (hunk 2 above)
Comment (CraigglesO) [reply CHANGELOG.md:12]: Hmm, before I update, are you ok with undefined AND null? As I said below I noticed the test cases use `undefined`.

Comment (HarelM) [reply CHANGELOG.md:12]: I think calling setTerrain() is meaningless, but since you and another person thought differently, I might be wrong.
If you read my comments and still thinks calling se terrain without an argument is a good API feel free to push this forward.

**Comment by HarelM** [comment]
If the API enforces null, the tests should be changed according to the API.

**Review by HarelM — APPROVED** [review]

**Comment by neodescis** [comment]
FWIW, null is consistent with the rest of the API, whereas undefined is not.
